docs/querying/functions.md

```diff
@@ -432,6 +432,23 @@ by the number of seconds under the specified time range window, and should be
 used primarily for human readability.  Use `rate` in recording rules so that
 increases are tracked consistently on a per-second basis.
 
+## `info()`
+
+**This function has to be enabled via the [feature flag](../feature_flags.md#experimental-promql-functions) `--enable-feature=promql-experimental-functions`.**
+
+For each time series in `v`, `info(v instant-vector, [data-label-selector string])` finds 
+all `target_info` series with corresponding identifying labels (`instance` and `job`),
+and adds the union of their data (i.e., non-identifying) labels to the time series.
+`target_info`'s data labels correspond to OpenTelemetry resource attributes.
+The optional second argument allows specifying through label matchers, which data labels
+should be picked. Example: `{k8s_container_name=~".+"}`. Through this argument, you can
+also specify the info series to consider, by providing a `__name__` label matcher, e.g.
+`{__name__="target_info"}`. For the time being however, only `target_info` will be considered.
+The set of time series set augmented with any matching info series labels gets returned.
+
+In the future, `info` will recognize all info metrics, not just `target_info`, and also detect their 
+respective identifying labels.
```

Comment:
Just to be clear, I don't mean to block anything if the team has already decided the future of info :)

Reply:
It basically refers to what @beorn7 said above:

> only works for target_info, not for info metrics in general, however we will define what an info metric is in the future; and only considers job and instance as identifying labels, however we will define what the identifying labels are in the future

It's very difficult to say at this point whether info metrics will be made obsolete by metadata support in the future, or whether it would make sense to improve support for them.

Reply:
As said, this is an experiment to find out how useful the info function works for target_info. If it is promising, we can later experiment with how useful it is for the many already existing info metrics. And in parallel, we can see how metadata handling evolves. It can well be that we decide to not need the info function because we have solved the problem with metadata handling. It could also be that the info function is just the right way of accessing (some parts of the) metadata in a a way that is in line with the current Prometheus data model.